zulip_bots/zulip_bots/bots/baremetrics/baremetrics.py

```diff
@@ -0,0 +1,171 @@
+            return
+
+        response = self.generate_response(message['content'])
+        bot_handler.send_reply(message, response)
+
+    def generate_response(self, command: str) -> str:
+        try:
+            if command.lower() == 'account-info':
+                return self.get_account_info()
+
+            if command.lower() == 'list-sources':
+                return self.get_sources()
+
+            part_commands = command.split()
+
+            try:
+                if part_commands[0].lower() == 'list-plans':
+                    return self.get_plans(part_commands[1])
+
+                if part_commands[0].lower() == 'list-customers':
+                    return self.get_customers(part_commands[1])
+
+                if part_commands[0].lower() == 'list-subscriptions':
+                    return self.get_subscriptions(part_commands[1])
+
+            except IndexError:
+                return 'Missing Params.'
+        except KeyError:
+            return 'Invalid Response From API.'
+
+        return 'Invalid Command.'
+
+    def get_account_info(self) -> str:
+        url = "https://api.baremetrics.com/v1/account"
+        account_response = requests.get(url, headers=self.auth_header)
+
+        account_data = account_response.json()
+        account_data = account_data['account']
+
+        response = '**Your account information:** \n'
+        response += 'Id: {id}\n'.format(id=account_data['id'])
+        response += 'Company: {company}\n'.format(company=account_data['company'])
+        response += 'Default Currency: {currency_name}'.format(currency_name=account_data['default_currency']['name'])
+
+        return response
+
+    def get_sources(self) -> str:
+        url = 'https://api.baremetrics.com/v1/sources'
+        sources_response = requests.get(url, headers=self.auth_header)
+
+        sources_data = sources_response.json()
+        sources_data = sources_data['sources']
+
+        response = '**Listing sources:** \n'
+        for index, source in enumerate(sources_data):
+            response += '{}.ID: {}\nProvider: {}\nProvider ID: {}\n\n'.format(index + 1, source['id'],
+                                                                              source['provider'],
+                                                                              source['provider_id'])
+
+        return response
+
+    def get_plans(self, source_id: str) -> str:
+        url = 'https://api.baremetrics.com/v1/{}/plans'.format(source_id)
+        plans_response = requests.get(url, headers=self.auth_header)
+
+        plans_data = plans_response.json()
+        plans_data = plans_data['plans']
+
+        response = '**Listing plans:** \n'
+        for index, plan in enumerate(plans_data):
+            response += '{}.Name: {}\nActive: {}\nInterval: {}\nInterval Count: {}\nAmounts: \n' \
+                .format(index + 1, plan['name'], plan['active'], plan['interval'], plan['interval_count'])
+
```

Comment:
I feel like this part of the code could be a bit more readable. Maybe change it to something like this:
```python
template = '{}.Name: {}\nActive: {}\nInterval: {}\nInterval Count: {}\nAmounts: \n'
response += template.format(index + 1, plan['name'], plan['active'], plan['interval'], plan['interval_count'])
```

That is a bit easier on the eyes and I'm not sure breaking up a function call over two lines in the way you did is very Pythonic in a PEP8 context or not. :)